### PR TITLE
feat(tls): added mTLS connection to clickhouse for uploaders

### PR DESCRIPTION
**Pull request by lordvidex**
Refs: #121

### CODE REVIEW

File: uploader/base.go [hunk 1]
```diff
@@ -250,9 +252,24 @@ func (u *Base) insertRowBinary(table string, data io.Reader) error {
 		return err
 	}
 
+	transport := &http.Transport{DisableKeepAlives: true}
+
+	if u.config.TLS != nil {
```
[the remark this answers is not in the document]
Comment (msaf1980) [reply uploader/base.go:257]: DisableKeepAlives is required ? Why ?

[the remark this answers is not in the document]
Comment (lordvidex) [reply uploader/base.go:257]: > 

do you mean checking the queryURL for `https://` prefix to determine whether to add TLS?

[the remark this answers is not in the document]
Comment (msaf1980) [reply uploader/base.go:257]: DisableKeepAlives is required ? Why ?

> > DisableKeepAlives is required ? Why ?
> 
> The old transport had DisableKeepAlives set to true, and I left it just for backwards compatibility

OK/

[the remark this answers is not in the document]
Comment (msaf1980) [reply uploader/base.go:257]: > > May be also add check for https ? May be not needed customized http.Client for non-TLS ?
> 
>     * do you mean checking the queryURL for `https://` prefix to determine whether to add TLS?
Yes. and may be run  ParseClientTLSConfig on build configure phase (add to structure param with name like TLSConfig and annotation tag  `toml:"-"`.
For check TLS config fatal error and abort if failure.
> 
>     * In the scenario for normal http connection, we still have to include the `Timeout` in the http client.
Agree

File: README.md [hunk 2]
```diff
@@ -239,14 +256,14 @@ concat = "."
...
 enabled = false
 
-# You can use tag matching like in InfluxDB. Format is exactly the same.
-# It will parse all metrics that don't have tags yet.
-# For more information see https://docs.influxdata.com/influxdb/v1.7/supported_protocols/graphite/
-# Example:
+# you can use tag matching like in influxdb. format is exactly the same.
+# it will parse all metrics that don't have tags yet.
+# for more information see https://docs.influxdata.com/influxdb/v1.7/supported_protocols/graphite/
+# example:
```
[the remark this answers is not in the document]
Comment (lordvidex) [reply README.md:266]: I can revert them, I think it's probably prettierd

[the remark this answers is not in the document]
Comment (msaf1980) [reply README.md:266]: > I can revert them, I think it's probably prettierd

May be revert ? All other comments start from upper letter.

**Pull request merged**